python/ray/tune/pbt.py

```diff
@@ -2,189 +2,248 @@
+            else:
+                new_config[key] = config[key] * 0.8
+    print(
+        "[explore] perturbed config from {} -> {}".format(config, new_config))
+    return new_config
+
+
+def make_experiment_tag(orig_tag, config, mutations):
+    """Appends perturbed params to the trial name to show in the console."""
+
+    resolved_vars = {}
+    for k in mutations.keys():
+        resolved_vars[("config", k)] = config[k]
+    return "{}@perturbed[{}]".format(orig_tag, _format_vars(resolved_vars))
 
 
 class PopulationBasedTraining(FIFOScheduler):
-    """Implements the Population Based Training algorithm as described in the
-    PBT paper (https://arxiv.org/abs/1711.09846)(Experimental):
+    """Implements the Population Based Training (PBT) algorithm.
+
+    https://deepmind.com/blog/population-based-training-neural-networks
+
+    PBT trains a group of models (or agents) in parallel. Periodically, poorly
+    performing models clone the state of the top performers, and a random
+    mutation is applied to their hyper-parameters in the hopes of
+    outperforming the current top models.
+
+    Unlike other hyperparameter search algorithms, PBT mutates hyperparameters
+    during training time. This enables very fast hyperparameter discovery and
+    also automatically discovers good annealing schedules.
+
+    This Ray Tune PBT implementation considers all trials added as part of the
+    PBT population. If the number of trials exceeds the cluster capacity,
+    they will be time-multiplexed as to balance training progress across the
+    population.
 
     Args:
-        time_attr (str): The TrainingResult attr to use for documenting length
-            of time since last ready() call. Attribute only has to increase
-            monotonically.
+        time_attr (str): The TrainingResult attr to use for comparing time.
+            Note that you can pass in something non-temporal such as
+            `training_iteration` as a measure of progress, the only requirement
+            is that the attribute should increase monotonically.
         reward_attr (str): The TrainingResult objective value attribute. As
-            with 'time_attr'. this may refer to any objective value that
-            is supposed to increase with time.
-        grace_period (float): Period of time, in which algorithm will not
-            compare model to other models.
-        perturbation_interval (float): Used in the truncation ready function to
-            determine if enough time has passed so that a agent can be tested
-            for readiness.
-        hyperparameter_mutations (dict); Possible values that each
-            hyperparameter can mutate to, as certain hyperparameters
-            only work with certain values.
+            with `time_attr`, this may refer to any objective value. Stopping
+            procedures will use this attribute.
+        perturbation_interval (float): Models will be considered for
```

Comment:
can you explicitly clarify here that this interval is in units of `time_attr`? It was pretty unclear, especially given the example being labeled "10 iterations"

Reply:
Fixed